src/uct/cuda/cuda_ipc/cuda_ipc_iface.c

```diff
@@ -447,12 +447,12 @@ static void uct_cuda_ipc_event_desc_init(ucs_mpool_t *mp, void *obj, void *chunk
 static void uct_cuda_ipc_event_desc_cleanup(ucs_mpool_t *mp, void *obj)
 {
     uct_cuda_ipc_event_desc_t *base = (uct_cuda_ipc_event_desc_t *) obj;
-    int active;
+    CUcontext cuda_context;
 
-    UCT_CUDADRV_CTX_ACTIVE(active);
-
-    if (active) {
-        UCT_CUDADRV_FUNC_LOG_ERR(cuEventDestroy(base->event));
+    UCT_CUDA_FUNC_LOG_ERR(cuCtxGetCurrent(&cuda_context));
+    if ((cuda_context != NULL) &&
+        (!memcmp(&cuda_context, &base->cuda_context, sizeof(cuda_context)))) {
+        UCT_CUDA_FUNC_LOG_ERR(cudaEventDestroy(base->event));
```

Comment:
seems like duplication of the code in cuda_copy

Reply:
will add macro